python/cudf/cudf/utils/cudautils.py

```diff
@@ -185,17 +186,34 @@ def find_first(arr, val, compare="eq"):
                 gpu_mark_found_int.forall(found.size)(
                     arr, val, found, arr.size
                 )
-    from cudf.core.column import as_column
 
-    found_col = as_column(found)
+    from cudf.core.column.column import as_column
+
+    return as_column(found).set_mask(mask)
```

Comment:
@kkraus14 I am using the mask here.

Reply:
Looks like I just commented too quickly :)

Reply:
No, I was too late.